components/automate-ui/src/app/pages/automate-settings/automate-settings.component.spec.ts

```diff
@@ -88,78 +130,173 @@ describe('AutomateSettingsComponent', () => {
+          genNestedIngestJob('infra', 'periodic_purge_timeseries', 'actions', 2, false)],
 
-      const deleteMssingNodesValues = component.automateSettingsForm
-        .controls.deleteMissingNodes.value;
+      // Services --> not yet enabled
+      // ['serviceGroupNoHealthChecks'],
+      // ['serviceGroupRemoveServices'],
 
-      expect(deleteMssingNodesValues.disable).toEqual(false);
-      expect(deleteMssingNodesValues.threshold).toEqual('24');
-      expect(deleteMssingNodesValues.unit).toEqual('h');
-    });
+      // Client Runs
+      ['clientRunsRemoveData', 'non-nested', 'not applicable',
+          genInjestJob('infra', 'missing_nodes', '5m', false)],
+      ['clientRunsLabelMissing', 'non-nested', 'not applicable',
+          genInjestJob('infra', 'missing_nodes_for_deletion', '6h', false)],
+      ['clientRunsRemoveNodes', 'nested', 'converge-history',
+          genNestedIngestJob('infra', 'periodic_purge_timeseries', 'converge-history', 7, false)],
 
-    it('does not updates the "eventFeed" form group', () => {
-      component.updateForm(mockJobSchedulerStatus);
+        // Compliance
+      ['complianceRemoveReports', 'nested', 'compliance-reports',
+          genNestedIngestJob('compliance', 'periodic_purge', 'compliance-reports', 8, false)],
+      ['complianceRemoveScans', 'nested', 'compliance-scans',
+          genNestedIngestJob('compliance', 'periodic_purge', 'compliance-scans', 9, false)]
+    ], function(formName: string, jobType: string, nestedName: string, job: IngestJob) {
+      it(`when updating ${formName} form,
+            the form data is extracted from the ${jobType} form`, () => {
+        const thisJobScheduler = new JobSchedulerStatus([job]);
+        component.updateForm(thisJobScheduler);
 
-      const eventFeedValues = component.automateSettingsForm
-        .controls.eventFeed.value;
+        const newFormValues = component[formName].value;
 
-      // These are the defaults
-      expect(eventFeedValues.disable).toEqual(false);
-      expect(eventFeedValues.threshold).toEqual('');
-      expect(eventFeedValues.unit).toEqual('d');
+        let jobData = job;
+        if (jobType === 'nested') {
+          jobData = job.purge_policies.elasticsearch.find(item => item.name === nestedName);
+          expect(newFormValues.threshold).toEqual(jobData.older_than_days);
+        } else {
+          // non-nested threshold is stored differently, so we need to separate it
+          // into threshold and unit first.
+          const [jobThreshold, jobUnit] = [job.threshold.slice(0, job.threshold.length - 1),
+                                           job.threshold.slice(-1)
+          ];
+          expect(newFormValues.threshold).toEqual(jobThreshold);
+          expect(newFormValues.unit).toEqual(jobUnit);
+        }
+
+        // all forms share the same expectation of being disabled
+        expect(newFormValues.disabled).toEqual(jobData.disabled);
+      });
     });
```

Comment:
@msorens - before I go further on this, do you think it makes sense to separate out non-nested and nested forms into their own separate test blocks for clarity?  I'm using an `if/else` inside this test block and I feel like this is going more difficult for others to read.

Reply:
Great question, and yes I do. Should strive for cyclomatic complexity === 1  for test cases whenever possible. (That may or may not mean anything to you; if not, happy to chat!)